src/main/java/com/atherys/towns/service/TownRaidService.java

```diff
@@ -0,0 +1,187 @@
+package com.atherys.towns.service;
+
+import com.atherys.towns.AtherysTowns;
+import com.atherys.towns.TownsConfig;
+import com.atherys.towns.facade.TownFacade;
+import com.atherys.towns.facade.TownRaidFacade;
+import com.atherys.towns.facade.TownsMessagingFacade;
+import com.atherys.towns.model.RaidPoint;
+import com.atherys.towns.model.entity.Town;
+import com.atherys.towns.util.MathUtils;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.spongepowered.api.boss.BossBarColors;
+import org.spongepowered.api.boss.BossBarOverlays;
+import org.spongepowered.api.boss.ServerBossBar;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.entity.Transform;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.scheduler.Task;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Singleton
+public class TownRaidService {
+    @Inject
+    TownsConfig config;
+
+    @Inject
+    TownService townService;
+
+    @Inject
+    TownRaidFacade townRaidFacade;
+
+    @Inject
+    TownFacade townFacade;
+
+    @Inject
+    TownsMessagingFacade townsMsg;
+
+    Map<UUID, RaidPoint> activeRaids = new ConcurrentHashMap<>();
+
+    public TownRaidService() {
+
+    }
+
+    private void removeEntityFromWorld(World world, UUID entityUUID) {
+        Optional<Entity> entity = world.getEntity(entityUUID);
+        entity.ifPresent(Entity::remove);
+    }
+
+    public void removeRaidPoint(UUID id) {
+        RaidPoint point = activeRaids.get(id);
+        World world = point.getPointTransform().getExtent();
+
+        //First remove players from BossBar
+        ServerBossBar bossBar = point.getRaidBossBar();
+        bossBar.removePlayers(bossBar.getPlayers());
+
+        //Next Remove RaidPoint Entity
+        removeEntityFromWorld(world, id);
+
+        //Finally, Remove Particles
+        point.getParticleUUIDs().forEach(uuid -> removeEntityFromWorld(world, uuid));
+
+        activeRaids.remove(id);
+    }
+
+    public RaidPoint getRaidPoint(UUID id) {
+        return activeRaids.get(id);
+    }
+
+    public boolean isIdRaidEntity(UUID entityId) {
+        return activeRaids.values().stream().anyMatch(point -> point.getRaidPointUUID().equals(entityId));
```

Comment:
Can't we just check if `activeRaids` has the entity UUID as a key? The UUID of the entity appears to be the key of the map

Reply:
Ah yes, that was left over from when I was doing something else with the HashMap, I will fix that.